src/sentry/static/sentry/app/views/integrationPipeline/pipelineView.tsx

```diff
@@ -10,28 +10,37 @@ import AwsLambdaProjectSelect from './awsLambdaProjectSelect';
  * This component is a wrapper for specific pipeline views for integrations
  */
 
-const pipelineMapper = {
-  awsLambdaProjectSelect: AwsLambdaProjectSelect,
-  awsLambdaFunctionSelect: AwsLambdaFunctionSelect,
-  awsLambdaFailureDetails: AwsLambdaFailureDetails,
+const pipelineMapper: Record<string, [React.ElementType, string]> = {
+  awsLambdaProjectSelect: [AwsLambdaProjectSelect, 'Select Project'],
+  awsLambdaFunctionSelect: [AwsLambdaFunctionSelect, 'Select Lambdas'],
+  awsLambdaFailureDetails: [AwsLambdaFailureDetails, 'View Failures'],
 };
 
 type Props = {
   pipelineName: string;
   [key: string]: any;
 };
 
-const PipelineView = (props: Props) => {
-  const {pipelineName, ...rest} = props;
-  const Component = pipelineMapper[pipelineName];
-  if (!Component) {
-    throw new Error(`Invalid pipeline name ${pipelineName}`);
+export default class PipelineView extends React.Component<Props> {
+  componentDidMount() {
+    // update the title based on our mappings
+    const title = this.mapping[1];
+    document.title = title;
```

Comment:
@scefali where is this `document` from?

Reply:
@NisanthanNanthakumar Technically it's `window.document`: https://developer.mozilla.org/en-US/docs/Web/API/Window/document

Reply:
@scefali ahh ok!